static/js/protein/chart.ts

```diff
@@ -14,40 +14,151 @@
  * limitations under the License.
  */
 
-import * as d3 from "d3";
+ import * as d3 from "d3";
 
-const SVGNS = "http://www.w3.org/2000/svg";
-const XLINKNS = "http://www.w3.org/1999/xlink";
-
-const HEIGHT = 224;
-const WIDTH = 500;
-
-/**
- * Draw bar chart for tissue score.
- */
-export function drawTissueScoreChart(
-  id: string,
-  data: { name: string; value: string }[]
-): void {
-  // TODO: convert data to tmpData format.
-  const tmpData = [{ x1: 20, x2: 60, y1: 30, y2: 50 }];
-  const svg = d3
-    .select("#" + id)
-    .append("svg")
-    .attr("xmlns", SVGNS)
-    .attr("xmlns:xlink", XLINKNS)
-    .attr("width", WIDTH)
-    .attr("height", HEIGHT);
-
-  svg
-    .append("g")
-    .selectAll("rect")
-    .data(tmpData)
-    .enter()
-    .append("rect")
-    .attr("x", (d) => d.x1)
-    .attr("y", (d) => d.y1)
-    .attr("width", (d) => d.x2 - d.x1)
-    .attr("height", (d) => d.y2 - d.y1)
-    .attr("fill", "teal");
-}
+ const SVGNS = "http://www.w3.org/2000/svg";
+ const XLINKNS = "http://www.w3.org/1999/xlink";
+ 
+ const HEIGHT = 224;
+ const WIDTH = 500;
+ 
+ /**
+  * Draw bar chart for tissue score.
+  */
+ export function drawTissueScoreChart(
+   id: string,
+   data: { name: string; value: string }[]
+ ): void {
+     const tmpData = [
+       { name: "Cartilage", value: 3 },
+       { name: "Eye", value: 2 },
+       { name: "Liver", value: 1 },
+       { name: "Skin", value: 0 },
+     ];
+     const size = Object.keys(tmpData);
+     const barHeight = 35;
+     const margin = { top: 70, right: 50, bottom: 10, left: 50 };
+     const height = 400 - margin.top - margin.bottom;
+     const width = 460 - margin.left - margin.right;
+     const x = d3.scaleLinear()
+         .domain([0, d3.max(tmpData, (d) => d.value)])
+         .range([margin.left, width - margin.right]);
+     const y = d3.scaleBand()
+         .domain(size)
+         .rangeRound([margin.top, height - margin.bottom])
+         .padding(0.1);
+     function formatTick(d) {
+         if (d == 0) {
+             return "NotDetected";
+         } else if (d == 1) {
+             return "Low";
+         } else if (d == 2) {
+             return "Medium";
+         } else {
+             return "High";
+         }
+         };
+       const svg = d3.selectAll("#" + id).append('svg')
+         .attr("xmlns", SVGNS)
+         .attr("xmlns:xlink", XLINKNS)
+         .attr("width", WIDTH)
+         .attr("height", HEIGHT);
+         //.attr("viewBox", [0, 0, width, height]);
+     
+     svg.append("g")
+         .selectAll("rect")
+         .data(tmpData.sort((a, b) => d3.descending(a.value, b.value)))
+         .enter()
+         .append("rect")
+         .attr("x", x(0))
+         .attr("y", (d) => y(d.name)) //y(i)
```

Comment:
this is different from the observable code:

```
      .attr("y", (d, i) => y(i) + y.bandwidth() / 2)
```

this line sets the y position for the rectangle, so i'd check here. you could:
- manually return different values
- add print statements: `(d) => { console.log(y(d.name)), return y(d.name); }`